src/main/java/com/coveros/selenified/element/check/Equals.java

```diff
@@ -0,0 +1,370 @@
+            return null;
+        }
+        // get the selected value
+        String elementValue = getElement().get().selectedValue();
+        // record the result
+        if (!elementValue.equals(expectedValue)) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_VALUE + elementValue + "</b>", timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + HAS_VALUE + elementValue + "</b>", timeTook, Success.PASS);
+        }
+        return elementValue;
+    }
+
+    /**
+     * Verifies that the element's select options equal the provided expected
+     * options. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedOptions the expected input value of the element
+     */
+    void selectOptions(String... expectedOptions);
+
+    default String[] checkSelectOptions(String[] expectedOptions, double waitFor, double timeTook) {
+        // record the action, and verify it's a select
+        if (!isPresentSelect(
+                getElement().prettyOutput() + " with select options of <b>" + String.join("</b>, <b>" + expectedOptions) + "</b>", waitFor)) {
+            return null;
+        }
+        // get the actual select options
+        String[] elementOptions = getElement().get().selectOptions();
+        // record the result
+        if (!Arrays.toString(elementOptions).equals(Arrays.toString(expectedOptions))) {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + OPTIONS + Arrays.toString(elementOptions) + "</b>",
+                    timeTook, Success.FAIL);
+        } else {
+            getOutputFile().recordActual(getElement().prettyOutputStart() + OPTIONS + Arrays.toString(elementOptions) + "</b>",
+                    timeTook, Success.PASS);
+        }
+        return elementOptions;
+    }
+
+    /**
+     * Verifies that the element's select values equal the provided expected
+     * values. If the element isn't present or a select, this will constitute a
+     * failure, same as a mismatch. This information will be logged and
+     * recorded, with a screenshot for traceability and added debugging support.
+     *
+     * @param expectedValues the expected input value of the element
+     */
+    void selectValues(String... expectedValues);
+
+    default String[] checkSelectValues(String[] expectedValues, double waitFor, double timeTook) {
+        // record the action, and verify it's a select
+        if (!isPresentSelect(
+                getElement().prettyOutput() + " with select values of <b>" + Arrays.toString(expectedValues) +
+                        "</b>", waitFor)) {
+            return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/sonar/coding_rules#rule_key=squid%3AS1168)